spirv_glsl.cpp

```diff
@@ -2802,7 +2802,9 @@ void CompilerGLSL::emit_texture_op(const Instruction &i)
 	auto &imgtype = get<SPIRType>(type.self);
 
 	// Mark that this shader reads from this image
-	imgtype.image.is_read = true;
+	auto *p_var = maybe_get_backing_variable(img);
+	if (p_var)
+		unset_decoration(p_var->self, DecorationNonReadable);
```

Comment:
Ok, I think I see why ... This is used because OpImageRead is implemented through a normal fetch in MSL.

It is a bit confusing to have this code here, could you just move it to OpImageRead case statement in spirv_msl.cpp instead?